File: java-kata/src/test/java/org/xpdojo/bank/MoneyTest.java
package org.xpdojo.bank;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;
import static org.xpdojo.bank.Money.amountOf;

@DisplayName("With Money we can ... ")
public class MoneyTest {

    @Test
    public void addTwoAmountsTogether() {
        assertThat(amountOf(10.0).add(amountOf(15.0))).isEqualTo(amountOf(25.0));
    }

    @Test
    public void reductTwoAmounts() {
        assertThat(amountOf(90.0).less(amountOf(20.0))).isEqualTo(amountOf(70.0));
    }
}

File: kotlin-kata/src/test/kotlin/org/xpdojo/bank/MoneyTest.kt
package org.xpdojo.bank

import io.kotest.matchers.shouldBe
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test
import org.xpdojo.bank.Money.Companion.amountOf

@DisplayName("With Money we can ...")
class MoneyTest {

    @Test
    fun `add two monies to create a new money`() {
        (amountOf(12.0) add amountOf(34.0)) shouldBe amountOf(46.0)
    }
}